Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java

```diff
@@ -0,0 +1,130 @@
+package edu.asu.spring.quadriga.web.publicwebsite;
+
+/**
+ * This controller has all the mappings required to view the statistics of the project
+ * 
+ * @author ajaymodi
+ *
+ */
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import javax.xml.bind.JAXBException;
+
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import edu.asu.spring.quadriga.domain.IConceptStats;
+import edu.asu.spring.quadriga.domain.network.INetwork;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.network.INetworkManager;
+import edu.asu.spring.quadriga.service.network.impl.INetworkTransformationManager;
+import edu.asu.spring.quadriga.service.publicwebsite.impl.ProjectStats;
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+
+@PropertySource(value = "classpath:/user.properties")
+@Controller
+public class ProjectStatsController {
+
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	@Autowired
+	private INetworkManager networkmanager;
+
+	@Autowired
+	private INetworkTransformationManager transformationManager;
+
+	@Autowired
+	private ProjectStats ps;
+
+	public IRetrieveProjectManager getProjectManager() {
+		return projectManager;
+	}
+
+	private IProject getProjectDetails(String name)
+	        throws QuadrigaStorageException {
+		return projectManager.getProjectDetailsByUnixName(name);
+	}
+
+	private JSONArray getProjectStatsJson(List<IConceptStats> gtc)
+	        throws JSONException {
+		JSONArray obj = new JSONArray();
+
+		int cnt = System.getProperty("topCount") == null ? 0 : Integer

```

Comment:
Your properties are not really system properties and I wouldn't read them like that. Choose one of the options here:http://www.mkyong.com/spring/spring-propertysources-example/ 
(```@Value``` annotation or autowire an ```Environment```)